src/Mutex/Impl/Internal/MutexWorkflowInput.cs

```diff
@@ -0,0 +1,3 @@
+namespace TemporalioSamples.Mutex.Impl.Internal;
+
+internal record MutexWorkflowInput(IEnumerable<LockRequest> InitialRequests);
```

Comment:
```suggestion
internal record MutexWorkflowInput(IReadOnlyCollection<LockRequest> InitialRequests);
```

I have been discouraging enumerable as parameters in this use case because of the confusion with users when they expect it to be lazy or not re-enumerable.